valora-inc/wallet
# fix(fiatconnect): error banner about providers fetch failing in some cases

### Description

With the move to Home Screen Actions, the path triggering fetching FiatConnect providers changed and choosing "Add" from the home screen, would should an error banner the first time when asked to select providers. See Slack [thread](https://valora-app.slack.com/archives/C04B61SJ6DS/p1695131047003169).

This PR moves the logic to fetch providers early to the `FiatExchangeCurrency` screen instead, which is common to all CiCo flows. Note that this will cause the refetch of providers to be a bit more frequent. Let me know if this could cause issues.

Also `_getQuotes` now tries to fetch providers if `null` to prevent this bug from coming back if we ever change the navigation again and cause the early fetch not to be triggered.

### Test plan

- Updated unit tests
- Updated e2e tests
- Tested manually that the error banner is not displayed anymore on step 6 of the following steps:
  1. install a fresh new Valora
  2. restore an account which already has funds
  3. tap the “Add” button from the home screen”
  4. select cUSD
  5. enter an amount, for instance 10
  6. notice the error message displayed

### Related issues

- See Slack [thread](https://valora-app.slack.com/archives/C04B61SJ6DS/p1695131047003169).
- Fixes RET-843

### Backwards compatibility

Yes
